cmd/ncproxy/ncproxy.go

```diff
@@ -24,8 +24,29 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+type computeAgentCache struct {
+	// lock for synchronizing read/write access to `cache`
+	rw *sync.RWMutex
+	// mapping of container ID to shim compute agent ttrpc service
+	cache map[string]computeagent.ComputeAgentService
+}
+
+func newComputeAgentCache() *computeAgentCache {
+	return &computeAgentCache{
+		cache: make(map[string]computeagent.ComputeAgentService),
+	}
+}
+
 // GRPC service exposed for use by a Node Network Service.
-type grpcService struct{}
```

Comment:
Side note: I would love if in the future we can find better names for the "network agent facing" and "shim facing" services than `grpcService` and `ttrpcService`.